R/read_csv.R

```diff
@@ -189,25 +194,42 @@ read_cmdstan_csv <- function(files,
       all_draws <- metadata$output_samples
     } else if (metadata$method == "optimize") {
       all_draws <- 1
-    }
-    suppressWarnings(
-      draws <- vroom::vroom(
-        output_file,
-        comment = "#",
-        delim = ',',
-        trim_ws = TRUE,
-        col_select = col_select,
-        col_types = c("lp__" = "d"),
-        altrep = FALSE,
-        progress = FALSE,
-        skip = metadata$lines_to_skip,
-        n_max = all_draws * 2
+    }  
+    
+    if (metadata$method == "generate_quantities") {
+      col_types = list()
+      # set the first arg as double
+      # to silence the type detection info
+      col_types[[col_select[1]]] = "d"
+      suppressWarnings(      
+        draws <- vroom::vroom(
+          output_file,
+          comment = "#",
+          delim = ',',
+          col_select = col_select,
+          col_types = col_types,
+          trim_ws = TRUE,
+          altrep = FALSE,
+          progress = FALSE,
+          skip = metadata$lines_to_skip)
       )
-    )
-    if (ncol(draws) == 0) {
-      stop("The supplied csv file does not contain any data!", call. = FALSE)
```

Comment:
This error can never happen. If there are no CSV params it will fail earlier. If there are at least param names it returns an empty draws array.